### PR TITLE
feat: make atracker + cos opt-in feature

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
This PR updates the root of the module to make the atracker route and bucket for log storage optional based on a variable provided by the consumer

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: tests/pr_test.go [hunk 1]
```diff
@@ -77,3 +77,20 @@ func TestRunUpgradeDA(t *testing.T) {
 		assert.NotNil(t, output, "Expected some output")
 	}
 }
+
+func TestRunDAWithObs(t *testing.T) {
```
Comment (ocofaigh) [inline tests/pr_test.go:81]: No need to create a new test. Just set `provision_atracker_cos` to true in the current test

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@MatthewLemmond left one comment. Can you also maybe update some of the variable descriptions, as some of them are now only required if `provision_atracker_cos` is set to true.

**Comment by ocofaigh** [comment]
@MatthewLemmond You prob need to skip upgrade test:
```
│ Error: Value for undeclared variable
│ 
│ A variable named "provision_atracker_cos" was assigned on the command line,
│ but the root module does not declare a variable of that name. To use this
│ value, add a "variable" block to the configuration.
╵}
```

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**